src/Types/StructureType.php

```diff
@@ -48,9 +49,9 @@ public function addField($field, Type $type): void
 	/**
 	 * @param int|string $field
 	 */
-	public function overwriteInvalidField($field, Type $type): void
+	public function overwriteInvalidField($field, WithTypeAndValue $typeAndValue): void
 	{
-		$this->fields[$field] = $type;
+		$this->fields[$field] = $typeAndValue->getInvalidType();
```

Comment:
`WithTypeAndValue` should be stored in fields instead of getting `Type` here

Reply:
Hm, that's gonna break something. For example `addField` would have to accept `WithTypeAndValue` as well. Or we could wrap it.

Reply:
Imho we have to change api for obtaining errors from types anyway. Current one counts only with types, not values.

Reply:
So no changing of `addField`,  errors will be stored separately

Reply:
Oke, fine :)

Reply:
`addError` should probably change as well.

Reply:
Yup. It's just special case for callbacks, should have exception available too